training/xgboost/structured/base/requirements.txt

```diff
@@ -0,0 +1,8 @@
+# The pip syntax below allows us to not repeat
+# In order to not maintain two separate dependency
+# lists in setup.py vs requirements.txt
+# See https://caremad.io/posts/2013/07/setup-vs-requirement/
+
+--index-url https://pypi.python.org/simple/
```

Comment:
I haven't seen this before, does this work? What is it installing?

Reply:
Yes, as described in the link above it. It basically point to `setup.py` and runs it. So, these two become equivalent:
`pip install -r requirements.txt`
and
`python setup.py install`